doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+.. _conv_arithmetic:
+
+======================
+Convolution arithmetic
+======================
+
+The relationship between a convolution operation's input shape, kernel size,
+stride, padding and its output shape can be confusing at times.
+
+This tutorial aims at providing an intuitive understanding of how the different
+convolution parameters influence the shape of the output it produces.
+
+Quick reference
+===============
+
+Convolution
+-----------
+
+If a convolution is specified by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+then the output size is given by
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Transposed convolution
+----------------------
+
+If a transposed convolution is specified by
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+then the output size is given by
+
+.. math::
+
+    o = s (i - 1) + a + k - 2p, \quad a \in \{0, \ldots, s - 1\}
+
+where :math:`a` is a user-specified quantity used to distinguish between the
+:math:`s` different possible output sizes.
+
+Refresher: discrete convolution
+===============================
+
+The bread and butter of neural networks is affine transformations: a vector is
+received as input and is multiplied with a matrix to produce an output (to which
+a bias vector is usually added before passing the result through a
+nonlinearity). This is applicable to any type of input, be it an image, a sound
+clip or an unordered collection of features: their representation can always be
+flattened into a vector before the transformation.
+
+In the case of images and sound clips, there is some ordering (spatial or
+temporal) that is thrown away when they are flattened. This information may
+prove very handy in solving some tasks, like computer vision and speech
+recognition.
+
+These sorts of inputs share important properties:
+
+* They are stored as N-dimensional arrays (called *tensors* in Theano
+  terminology).
+* They feature one or more axes for which ordering matters (e.g. width and
+  height axes for an image, time axis for a sound clip).
+* One axis, the channel axis, is used to access different views of the data
+  (e.g. the red, green and blue channels of a color image, or the left and
+  right channels of a stereo audio track).
+
+A discrete convolution is a linear transformation that has the following
+properties:
+
+* It is sparse, i.e. only a few input units contribute to a given output unit.
+* Parameters are shared, i.e. the same transformation is applied to multiple
+  locations in the input.
+
+Here is an example of a discrete convolution:
+
+.. figure:: conv_arithmetic_figures/numerical_no_padding_no_strides.gif
+
+The blue grid is called the input *feature map*. We slide a *kernel* (dark blue)
+with value
+
+.. math::
+
+    \begin{pmatrix}
```

Comment:
I love this part! 
If it's not too much of a hussle, I think that having a static dark blue image here instead of the pmatrix would be even catchier.